### ISSUE TITLE
Add OSGi metadata

### ISSUE THREAD

**Issue by briantopping** [fommil/netlib-java#86]
Howdy, I came across your libs via ScalaNLP Breeze and was wondering if you would consider some help making them OSGi bundles. Such additions rarely require changes to the libs themselves and can be as easy as "export everything", thought some libraries have very clean interface and and implementation demarcations and it can be tuned quite nicely with very little effort.

**Comment by fommil** [comment]
Unfortunately making a release is exceptionally difficult due to the cross build nature. I don't think I'll be doing a release just for this, although if you were to send in a PR it word be part of the next release.

**Comment by fommil** [comment]
How does OSGi even handle natives?

**Comment by briantopping** [comment]
Hi Sam, thanks for the consideration. Timeframes wouldn't be a problem. I'm not sure about schedules on my end either.

I haven't build native libraries for OSGi before, but it seems to simplify deployment a lot. It appears that one jar can hold all the different platforms and build metadata helps sort it out on deployment. So maybe an n+1 native library for OSGi that has everything is an option. I'd play with it and we could try some different configurations on to see what feels best.

**Comment by fommil** [comment]
@briantopping I think you'd have to dive into the details of how a multi-platform cross compile of fortran libraries and JNI actually works we could say anything about packaging. I could add all the natives to one jar already with an extra step, but that doesn't help deployment because we're using JNILoader to simplify the cross platform loading --- indeed many people would complain that they were being forced to install natives for platforms that they don't care about. Good luck, this is one of the most necessarily complex builds you'll find :smile: Perhaps if it could be refactored into a build that can be build from a docker image, it would simplify things a lot but I can't cross compile Fortran libs for OSX.

**Comment by briantopping** [comment]
Yes, it's certainly formidable! I will have to plow through your custom plugin to get a better idea, but you did a really nice job with it all, kudos. Docker does sound interesting, but OS X support is weak.

**Comment by fommil** [comment]
#68 may need to be a dependency to anything to be honest.
